daemon/cluster/executor/container/controller.go

```diff
@@ -462,53 +459,33 @@ func (r *controller) Logs(ctx context.Context, publisher exec.LogPublisher, opti
 		}
 	)
 
-	brd := bufio.NewReader(rc)
 	for {
-		// so, message header is 8 bytes, treat as uint64, pull stream off MSB
-		var header uint64
-		if err := binary.Read(brd, binary.BigEndian, &header); err != nil {
-			if err == io.EOF {
-				return nil
-			}
-
-			return errors.Wrap(err, "failed reading log header")
+		msg, ok := <-msgs
```

Comment:
Hm, yeah, I overlooked that. I added a check. Because we now cancel the context, if we catch an error in the controller, we return instead of continuing.